src/evict/evict_lru.c

```diff
@@ -555,7 +562,17 @@ __evict_pass(WT_SESSION_IMPL *session)
 		    cache->bytes_dirty_intl + cache->bytes_dirty_leaf);
 
 		WT_RET(__evict_lru_walk(session));
-		WT_RET_NOTFOUND_OK(__evict_lru_pages(session, true));
+
+		/*
+		 * If the queue has been empty recently, keep queuing more
+		 * pages to evict.  If the rate of queuing pages is high
+		 * enough, this score will go to zero, in which case the
+		 * eviction server might as well help out with eviction.
+		 */
+		if (cache->evict_empty_score < WT_EVICT_EMPTY_SCORE_CUTOFF ||
+		    (!WT_EVICT_HAS_WORKERS(session) &&
+		    !__evict_queue_empty(cache->evict_urgent_queue)))
```

Comment:
This conditional is a bit confusing and could use more explanation in the comment.  If there are no eviction workers, and no pages on the urgent queue, the lone eviction server thread won't call `evict_lru_pages`?  Or is that situation solely dependent on the empty score getting low enough before it does so?